moonraker/components/power.py

```diff
@@ -1318,6 +1321,90 @@ async def _send_status_request(self) -> str:
         return "on" if resp["state"]["on"] else "off"
 
 
+# USB Hub Control
+# https://github.com/mvp/uhubctl
+# apt install uhubctl
+class UhubctlDevice(PowerDevice):
+    def __init__(self, config: ConfigHelper) -> None:
+        super().__init__(config)
+
+        self.bin_path = config.get("bin_path", "uhubctl")
+        self.hub, self.port = tuple(config.get("port").split("."))
+        self.initial_state = config.getboolean("initial_state", True)
+
+    def _uhubctl(self, args: List[str] = []) -> Tuple:
+        proc = subprocess.Popen(
+            [self.bin_path] + args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True
+        )
+
+        try:
+            out, err = proc.communicate(timeout=5)
+
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            out, err = proc.communicate()
+
+        return (out.strip().splitlines(), err.strip().splitlines())
+
+    def _port_status(self, toggle: str = "") -> List[str]:
+        result, err = self._uhubctl(
+            ["-l", self.hub, "-p", self.port] +
+            (["-a", toggle] if toggle else [])
+        )
+
+        if err:
+            logging.exception("uhubctl returned error: " + "\n".join(err))
+            return []
+
+        ports = {}
+
+        for line in result:
+            first_word = line.split(" ")[0]
+
+            if first_word in ["Current", "New"]:
+                hub = line.split(" ")[4]
+                continue
+
+            if first_word == "Sent":
+                continue
+
+            port, info = tuple(line.strip()[5:].split(": "))
+            ports[hub + "." + port] = info.split(" ")
+
+        id = self.hub + "." + self.port
+
+        if id in ports:
+            return ports[id]
+
+        return []
+
+    async def init_state(self) -> None:
+        await self.set_power("on" if self.initial_state else "off")
```

Comment:
This was added because I always want my printer to be off unless explicitly commanded to be on, but maybe there are other ways to toggle power devices automatically at startup?

Reply:
I recently made `initial_state` a global option for power devices (I just haven't updated the docs yet).  It functions as you suggest, the `initial_state` defaults to `None` if not configured, in which case we do not change the state of the device on init.  The change to your code should be relatively simple, remove line 1332 as the base class has already fetched the option, and change `init_state()` to:
```python
async def init_state(self) -> None:
    if self.initial_state is not None:
        await self.set_power("on" if self.initial_state else "off")
```